internal/table/table.go

```diff
@@ -140,44 +146,59 @@ func renderRow(providedColumns []string, colWidths []int) string {
 
 	result := ""
 
-	// Keep rendering lines until there's no column data left to render
-	for len(strings.Join(columns, "")) != 0 {
-		// Iterate over the columns by their line sizes
-		for n, maxLen := range colWidths {
-			// ignore columns that we do not have data for (they have been filled up with the last colValue already)
-			if len(columns) < n+1 {
-				continue
-			}
+	entries := make([][]colorize.Entry, len(columns))
 
-			colValue := []rune(columns[n])
+	widths := make([]int, len(colWidths))
+	copy(widths, colWidths)
+	for n, maxLen := range colWidths {
+		// ignore columns that we do not have data for (they have been filled up with the last colValue already)
+		if len(columns) < n+1 {
+			continue
+		}
 
-			// Detect multi column span
-			if len(colWidths) > n+1 && len(columns) == n+1 {
-				for _, v := range colWidths[n+1:] {
-					maxLen += v
-				}
+		// Detect multi column span
+		if len(colWidths) > n+1 && len(columns) == n+1 {
+			for _, v := range colWidths[n+1:] {
+				maxLen += v
 			}
+		}
 
-			maxLen = maxLen - (padding * 2)
+		maxLen = maxLen - (padding * 2)
+		widths[n] = maxLen
 
-			// How much of the colValue are we using for this line?
-			end := len(colValue)
-			if end > maxLen {
-				end = maxLen
-			}
+		entries[n] = colorize.GetCroppedText(columns[n], maxLen)
+	}
 
-			skipLineBreak := 0
-			if breakpos := runeSliceIndexOf(colValue, linebreakRune); breakpos != -1 && breakpos < end {
-				end = breakpos + 1
-				skipLineBreak = 1
-			}
+	lines := 0
+	for _, entry := range entries {
+		// Each entry represents a column. The longest column determines
+		// the number of rows the table will have.
+		if len(entry) > lines {
+			lines = len(entry)
+		}
+	}
```

Comment:
This logic still seems wrong, and a bit redundant. I think we should just iterate over `entries` until we reach an iteration where not a single column had data.

Reply:
As per our discussion yesterday the entires represent columns. I've updated the naming to (hopefully) be more clear. You are right though that this is redundant. I've updated `renderRow` and removed this loop.